src/appleseed/renderer/kernel/npr/nprclosures.h

```diff
@@ -0,0 +1,75 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2018 Esteban Tovagliari, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#ifndef APPLESEED_RENDERER_KERNEL_NPR_CLOSURES_H
+#define APPLESEED_RENDERER_KERNEL_NPR_CLOSURES_H
+
+// appleseed.renderer headers.
+#include "renderer/global/globaltypes.h"
+#include "renderer/modeling/input/inputarray.h"
+
+namespace renderer
+{
+
+//
+// NPRShading closure input values.
+//
+
+APPLESEED_DECLARE_INPUT_VALUES(NPRShadingInputValues)
+{
+};
+
+enum NPRContourFeatures
+{
+    ObjectInstanceID = 1 << 0,
+    MaterialID       = 1 << 1,
+    AllIDFeatures    = ObjectInstanceID | MaterialID,
+
+    OcclusionEdges   = 1 << 2,
+    CreaseEdges      = 1 << 3,
+    AllFeatures      = ~0
+};
+
+
+//
+// NPRContour closure input values.
+//
+
+APPLESEED_DECLARE_INPUT_VALUES(NPRContourInputValues)
+{
+    foundation::Color3f m_color;
+    float               m_opacity;
+    float               m_width;
+    float               m_occlusion_thereshold;
+    float               m_cos_crease_thereshold;
+    int32_t             m_features;
```

Comment:
You might want to use `int` here (which is the actual default type of enumerations).

If you really wanted to use a 32-bit signed integer, I would suggest to use `foundation::int32`. At some point we'll get rid of our typedefs and switch to `<cstdint>` types. We'll do that all at once.